### ISSUE TITLE
Central Window to Dispatch windows messages

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#3570]
**Reported by ragb on 2013-10-10 09:42**
As discussed on #1271 we need to process WM_DEVICECHANGE messages, so a rgistered window is needed. One is already created in the Freedom scientific driver and, per haps, elsewhere. So I purpose the creation of a central window where windows message handlers can be registered by various modules to be notified of WM_\* messages. Don't know if one can reuse the NVDA WX window for this or not.

A starting point might be this:

http://wiki.wxpython.org/MonitoringWindowsUsb

Although the code is somehow specific to windows monitoring, the generalized handling is there.

I'm not sure if we must block #1271 or not, because #1271 implementation will be far more elegant with this.

**Comment by bhavyashah** [comment]
@jcsteh's https://github.com/nvaccess/nvda/issues/3570#issuecomment-155310127 and https://github.com/nvaccess/nvda/issues/3570#ref-commit-2096318 seems to indicate that there has been work done on this. Could we please be updated about the present status of this work?

**Comment by jcsteh** [comment]
2096318 implements `windowUtils.CustomWindow`, which satisfies the narrowed requirements outlined in https://github.com/nvaccess/nvda/issues/3570#issuecomment-155310126.

Note that a4a47001 does actually implement a core custom window, though you can't register to receive arbitrary messages. However, given the concerns outlined in https://github.com/nvaccess/nvda/issues/3570#issuecomment-155310126, I still don't think going this far is a good idea anyway.

Regarding the original purpose of receiving device change messages (as outlined in https://github.com/nvaccess/nvda/issues/3570#issue-116036482), we could add some very simple logic into the core CustomWindow to catch WM_DEVICECHANGE and have this use an Action extension point (as provided by #7484) to notify interested components about device changes.

In short, I think we're done with this issue. Closing.
